src/sagemaker/workflow/airflow.py

```diff
@@ -41,8 +41,18 @@ def prepare_framework(estimator, s3_operations):
     elif estimator.uploaded_code is not None:
         bucket, key = s3.parse_s3_url(estimator.uploaded_code.s3_prefix)
     else:
+
+        # estimator.sagemaker_session._default_bucket may be None, unlike
+        # estimator.sagemaker_session.default_bucket.
+        # TODO: should this be updated to use default_bucket instead of _default_bucket? If yes,
```

Comment:
I dont see why not, we can try and let the integs test it. Nice find!

Reply:
cool, will try it out!

Reply:
updated and seems to be passing tests